src/FileHelper.php

```diff
@@ -513,40 +515,51 @@ private static function openDirectory(string $directory)
      *
      * If the path is a directory, any nested files/directories will be checked as well.
      *
-     * @param string ...$paths The directories to be checked.
+     * @param string[]|RecursiveDirectoryIterator[] $paths The directories to be checked.
      *
      * @throws LogicException If path is not set.
      *
-     * @return int Unix timestamp representing the last modification time.
+     * @return int|null Unix timestamp representing the last modification time.
      */
-    public static function lastModifiedTime(string ...$paths): int
+    public static function lastModifiedTime(string|RecursiveDirectoryIterator ...$paths): ?int
     {
         if (empty($paths)) {
             throw new LogicException('Path is required.');
         }
 
-        $times = [];
+        $time = null;
 
         foreach ($paths as $path) {
-            $times[] = self::modifiedTime($path);
+            if (is_string($path)) {
+                $timestamp = self::modifiedTime($path);
 
-            if (is_file($path)) {
-                continue;
+                if ($timestamp > $time) {
+                    $time = $timestamp;
+                }
+
+                if (is_file($path)) {
+                    continue;
+                }
+
+                $path = new RecursiveDirectoryIterator($path, FilesystemIterator::SKIP_DOTS);
             }
 
             /** @var iterable<string, string> $iterator */
             $iterator = new RecursiveIteratorIterator(
-                new RecursiveDirectoryIterator($path, FilesystemIterator::SKIP_DOTS),
+                $path,
                 RecursiveIteratorIterator::SELF_FIRST
             );
 
-            foreach ($iterator as $p => $_info) {
-                $times[] = self::modifiedTime($p);
+            foreach ($iterator as $path => $_info) {
+                $timestamp = self::modifiedTime($path);
```

Comment:
What happens if `RecursiveDirectoryIterator` is without option `SKIP_DOTS` ?

Maybe `modifiedTime()` must returns `int|null` also? `Null` for cases when filemtime returns `false`.

Reply:
> What happens if `RecursiveDirectoryIterator` is without option `SKIP_DOTS` ?
> 
> Maybe `modifiedTime()` must returns `int|null` also? `Null` for cases when filemtime returns `false`.

Interesting - it calculated filemtime from parent/root directory